modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/modals/CreateAPIApplicationModalContent.tsx

```diff
@@ -0,0 +1,306 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import {Text} from '@clayui/core';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import classNames from 'classnames';
+import {fetch, openToast} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {limitStringInputLengh, makeURLPathString} from '../utils/string';
+
+type Data = {
+	baseURL: string;
+	description: string;
+	title: string;
+	version: string;
+};
+
+type DataError = {
+	baseURL: boolean;
+	title: boolean;
+};
+
+interface HandleCreateInModal {
+	closeModal: voidReturn;
+	loadData: voidReturn;
+}
+
+const headers = new Headers({
+	'Accept': 'application/json',
+	'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+	'Content-Type': 'application/json',
+});
+
+export function CreateAPIApplicationModalContent({
+	closeModal,
+	loadData,
+}: HandleCreateInModal) {
+	const [data, setData] = useState<Partial<Data>>({});
+	const [displayError, setDisplayError] = useState<DataError>({
+		baseURL: false,
+		title: false,
+	});
+	const [userEditedURL, setUserEditedURL] = useState(false);
+
+	useEffect(() => {
+		for (const key in data) {
+			if (data[key as keyof Data] !== '') {
+				setDisplayError((previousErrors) => ({
+					...previousErrors,
+					[key]: false,
+				}));
+			}
+		}
+	}, [data]);
+
+	async function postData() {
+		fetch('http://localhost:8080/o/c/apiapplications', {
```

Comment:
Just to let you know that the endpoint is going to be changed. Check this https://github.com/liferay-headless/liferay-portal/pull/1284

Reply:
Nice! Also, this URL should be dynamic, not hard-coded. My bad :P